### PR TITLE
Basic Server

### PR DESCRIPTION
**Pull request by rayz**
Routes

GET / 

- returns all ToDos with id, description, completion status

POST /add

JSON Body
{
    "description": "..."
}

PUT /complete/{id}
- updates task with id to completed

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+func NewServer() *ToDoServer {
+	return &ToDoServer{
+		ToDoStore: NewToDoStore(),
+	}
+}
+
+func (server *ToDoServer) GetToDos(w http.ResponseWriter, req *http.Request) {
+	for _, task := range server.ToDoStore.Tasks {
+		fmt.Fprintf(w, "ID: %d\nTask: %s\nCompleted: %t\n", task.ID, task.Description, task.Completed)
```
Comment (scottopell) [inline server.go]: Future improvement here would be to support multiple [content-types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type), if none is specified, or if `text/plain` is specified, then this is a good response. But if `application/json` is requested, then return a json array containing each task.

File: server.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+type ToDoServer struct {
+	ToDoStore *ToDoStore
+	Count     int // total number of tasks (used as ID for individual task)
```
Comment (scottopell) [inline server.go]: doesn't appear to be used

File: store.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	Count int // total number of tasks (used as ID for individual task)
+}
+
+func NewToDoStore() *ToDoStore {
+	return &ToDoStore{}
+}
+
+func (store *ToDoStore) Add(task *Task) {
+	store.Count += 1
+	task.ID = store.Count
```
Comment (scottopell) [inline store.go:14]: Something to think about is how you would support deletions. If you use the count as an ID, you may run into issues depending on how you implement deletions.

File: server.go (hunk 2 above)
Comment (rayz) [reply server.go]: deleted

File: server.go (hunk 1 above)
Comment (rayz) [reply server.go]: noted. updated to return json array for now

File: store.go (hunk 3 above)
Comment (rayz) [reply store.go:14]: Maybe `Count` is misleading. It only ever increments so tasks should have unique ID

**Review by scottopell — APPROVED** [review]
Looks good to me!

**Pull request merged**